### PR TITLE
Reorganize doc flow by OS rather than by Task

### PR DESCRIPTION
**Pull request by curino**
Attempt to reorder docs by OS instead of by step. Maybe good maybe not.

### CODE REVIEW

File: documentation/01-Prerequisites.md [hunk 1]
```diff
@@ -6,76 +6,46 @@ These are one-time setup instructions that should be executed prior to following
...
-
-> Where `UbuntuVersion` can also be set to another supported version of Ubuntu.
-
-> Tip: you can also pass `--build-arg=http_proxy=http:/some-proxy-caching-host:3128` to help direct `apt` and `pip` to fetch the necessary packages via local caches.
-
-See [02-Build.md](./02-Build.md#docker) for instructions on how to run this image.
 
 ### Linux build tools
 
+#### Linux Build Manually
```
Comment (bpkroth) [inline documentation/01-Prerequisites.md:48]: This section isn't about building on Linux manually.
How about "Manual Linux Build Environment Setup" or something less verbose like "Manual Setup"

File: documentation/01-Prerequisites.md [hunk 2]
```diff
@@ -111,6 +81,113 @@ sudo apt-get install exuberant-ctags
 
 > When available `make ctags` can be invoked to help generate a `tags` database at the root of the source tree to allow easier code navigation in editors that support it.
 
+#### Docker
```
Comment (bpkroth) [inline documentation/01-Prerequisites.md:84]: Maybe too terse in this case?
"Create a docker image for building"

Comment (bpkroth) [inline documentation/01-Prerequisites.md:84]: Just above here, there is a path error:
`./script/...` should be `./scripts/...`

Do you want to fix that while you're in here or shall I make a separate PR?

File: documentation/01-Prerequisites.md [hunk 3]
```diff
@@ -6,76 +6,46 @@ These are one-time setup instructions that should be executed prior to following
...
-    - [Linux](#linux-python-install)
-    - [Windows](#windows-python-install)
-  - [Install Docker](#install-docker)
-    - [Linux Docker Install](#linux-docker-install)
+      - [Windows Build Manually](#win-build-manually)
+    - [Windows Pythoon Install](#windows-python-install)
     - [Windows Docker Install](#windows-docker-install)
+      
 
 > Note: Most Windows shell commands here expect `powershell` (or [`pwsh`](https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell-core-on-windows)).
```
Comment (bpkroth) [inline documentation/01-Prerequisites.md:31]: We should probably move that comment down by the Windows section instead.

File: documentation/01-Prerequisites.md [hunk 4]
```diff
@@ -369,3 +405,13 @@ As mentioned above, Docker on Windows first requires a Linux VM.
...
 
       From within this shell environment, `docker` cli commands should be proxied through to your `docker-dev-vm` prepared by `docker-machine`.
+
+
+
+--------------
+
+
+
+## Install build tools
```
Comment (bpkroth) [inline documentation/01-Prerequisites.md:415]: Dangling section

File: documentation/01-Prerequisites.md [hunk 5]
```diff
@@ -6,76 +6,46 @@ These are one-time setup instructions that should be executed prior to following
...
       - [Using Chocolatey](#using-chocolatey)
-      - [Manually](#manually)
-  - [Install Python Dependencies](#install-python-dependencies)
-    - [Linux](#linux-python-install)
-    - [Windows](#windows-python-install)
-  - [Install Docker](#install-docker)
-    - [Linux Docker Install](#linux-docker-install)
+      - [Windows Build Manually](#win-build-manually)
+    - [Windows Pythoon Install](#windows-python-install)
     - [Windows Docker Install](#windows-docker-install)
```
Comment (bpkroth) [inline documentation/01-Prerequisites.md:27]: Should see about having vscode automatically update this for us.  Might need an extension.  Perhaps one of these?
davidanson.vscode-markdownlint
shd101wyy.markdown-preview-enhanced

File: documentation/01-Prerequisites.md [hunk 6]
```diff
@@ -111,6 +81,113 @@ sudo apt-get install exuberant-ctags
...
+> Where `UbuntuVersion` can also be set to another supported version of Ubuntu.
+
+> Tip: you can also pass `--build-arg=http_proxy=http:/some-proxy-caching-host:3128` to help direct `apt` and `pip` to fetch the necessary packages via local caches.
+
+See [02-Build.md](./02-Build.md#docker) for instructions on how to run this image.
+
+
+### Clone the repository
+
+Cross platform
```
Comment (bpkroth) [inline documentation/01-Prerequisites.md:104]: This comment can probably be dropped.

File: documentation/01-Prerequisites.md [hunk 7]
```diff
@@ -111,6 +81,113 @@ sudo apt-get install exuberant-ctags
...
+    sudo apt-get -y install python3-pip python3.7-dev \
+        build-essential libfreetype-dev unixodbc-dev
+    ```
+
+    ```sh
+    python3.7 -m pip install --upgrade pip
+    python3.7 -m pip install setuptools
+
+    python3.7 -m pip install -r source/Mlos.Python/requirements.txt
+    ```
```
Comment (bpkroth) [inline documentation/01-Prerequisites.md:138]: Dunno if this should be addressed here or as a separate PR but #31, #45, and #43 all mention additional improvements in Python dependency setup.

File: documentation/01-Prerequisites.md [hunk 8]
```diff
@@ -6,76 +6,46 @@ These are one-time setup instructions that should be executed prior to following
 
 - [Prerequisites for building and using MLOS](#prerequisites-for-building-and-using-mlos)
   - [Contents](#contents)
-  - [Requirements](#requirements)
-  - [Clone the repository](#clone-the-repository)
-  - [Install build tools](#install-build-tools)
-    - [Docker](#docker)
+  
+  - [Linux](#linux)
```
Comment (bpkroth) [inline documentation/01-Prerequisites.md:10]: Should we add an OS X section as well, maybe just for Python and Docker?

File: documentation/01-Prerequisites.md (hunk 5 above)
Comment (bpkroth) [reply documentation/01-Prerequisites.md]: Nope it was joffreykern.markdown-toc or yzhang.markdown-all-in-one

File: documentation/01-Prerequisites.md (hunk 7 above)
Comment (bpkroth) [reply documentation/01-Prerequisites.md]: Discussed with @amueller earlier.  I think we're going to leave it here for now and fix it up afterwards.

For now I left a section for Conda install instructions as well.

**Comment by bpkroth** [comment]
@curino I finished another round of reorging this.  Can you talk a look please?

Will try and take a look at the Build instructions tomorrow as well.

**Comment by sergiy-k** [comment]
This is much easier to follow than before. Thank you!

**Review by sergiy-k — APPROVED** [review]

File: documentation/01-Prerequisites.md (hunk 8 above)
Comment (sergiy-k) [reply documentation/01-Prerequisites.md]: If we decide to add description for OS X then may be we should split the guide into separate files (one file for each OS) to keep each guide compact and focused? What do you think?

Comment (bpkroth) [reply documentation/01-Prerequisites.md]: Probably makes sense.  There's some redundancy then but so it goes.
One complication we're still working out is how best to keep relative links in the markdown files working so that browsing on github is natural (e.g. stays within tree, renders, etc.) but the github.io pages always take you back to master.
I think I have a hack to handle it.  Will work on it tomorrow.

Comment (curino) [reply documentation/01-Prerequisites.md]: I was about to start the OS X section, but shy away as I suspected I was not going to be able to finish it. Agreed on splitting if we add more than 2 OS
